File: backend/schemas/role.py
from sqlmodel import SQLModel, Field
from typing import List, Optional
from datetime import datetime

from .base import BaseResp


class CreateRole(SQLModel):
    role_name: str = Field(min_length=3, max_length=10)
    role_status: Optional[bool] = False
    role_desc: Optional[str] = Field(max_length=255)


class UpdateRole(SQLModel):
    id: int
    role_name: str
    role_status: Optional[bool]
    role_desc: Optional[str]


class RoleItem(SQLModel):
    id: int
    role_name: str
    role_status: Optional[bool]
    role_desc: Optional[str]
    create_time: datetime
    update_time: datetime


class RoleList(SQLModel):
    items: List[RoleItem]
    total: int

class RoleListResp(BaseResp):
    data: RoleList


class updateAccess(SQLModel):
    role_id: int
    access: List[int] = Field(default=[], description="权限集合")


class CreateAccess(SQLModel):
    access_name: str = Field(description="权限名称")
    scopes: str = Field(description="权限标识")
    parent_id: int = 0
    is_check: bool = False
    is_menu: bool = False

File: frontend/apps/web-ele/src/api/core/role.ts
import { requestClient } from '#/api/request';

export interface CreateRole {
  id?: number;
  role_name: string;
  role_status?: Boolean;
  role_desc?: string;
}

export interface RoleData {
  id?: number;
  role_name: string;
  role_status?: Boolean;
  role_desc?: string;
  create_time?: string;
  update_time?: string;
}

export interface RoleDataRes {
  items: RoleData[];
  total: number;
}

export interface MenuData {
  key: number;
  access_name: string;
  parent_id: number;
  scopes: string;
  access_desc: string;
  children?: MenuData[];
}

export interface MenuDataRes {
  all_access: MenuData[];
  role_access: number[];
}

export type MenuList = MenuData[];

export const reqAllRoleList = (page: number, limit: number, roleName: string) =>
  requestClient.get<RoleDataRes>('/role/list', {
    params: {
      size: limit,
      current: page,
      keyword: roleName,
    },
  });

export interface Role {
  value: number;
  label: string;
}
export type AllRole = Role[];

export interface UserRoleResp {
  all_role: AllRole;
  user_roles: number[];
}

export const reqAddOrUpdateRole = (data: RoleData) =>
  requestClient.post<RoleDataRes>('/role', data);

export const reqRemoveRole = (id: number) =>
  requestClient.delete<RoleDataRes>('/role', { params: { role_id: id } });

export const reqAllMenuList = (id: number) =>
  requestClient.get<MenuDataRes>('/access', { params: { role_id: id } });

export const reqSetPermission = (role_id: number, access: Number[]) =>
  requestClient.put('/access', { role_id, access });

export const reqUpdateUserRole = (id: number, data: number[]) =>
  requestClient.put('/user/set/role', { user_id: id, role_ids: data });

export const reqGetUserRole = (id: number) =>
  requestClient.get<UserRoleResp>('/role', { params: { user_id: id } });
